lib/plugin-loader.js

```diff
@@ -752,71 +756,275 @@ PluginLoader.prototype = {
     let newPlugins = [];
     let successCount = 0;
     const depgraph = new DependencyGraph(this.plugins);
-    for (const pluginDef of pluginDefs) {
-      if (pluginDef.error) {
-        //downstream will pick up error
-        newPlugins.push(pluginDef);
-      } else {
-        depgraph.addPlugin(pluginDef);
+    const componentCapabilities = await this.getComponentCapabilities(pluginContext.config); // Gets the environment and server information
```

Comment:
potentially this could be cached in the file for reuse, because i think (not certain) that if you do an app install when the server is already running, it will run through this again, so you get call at startup, then 1 call per each app added at runtime. could make it just 1 ever by storing the result.